### ISSUE TITLE
azurerm_app_service_environment wants to replace resources on tag updates

### ISSUE THREAD

**Issue by shawnferry** [hashicorp/terraform-provider-azurerm#5917]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Terraform (and AzureRM Provider) Version

* Terraform v0.12.20
* AzureRM provider.azurerm ~> 2.0

### Terraform Configuration Files

Any valid app service environment.

### Description / Feedback

When tags are updated 

```
# azurerm_app_service_environment.example must be replaced
-/+ resource "azurerm_app_service_environment" "example" {
...
      ~ tags                         = { # forces replacement
            "foo"           = "bar"
          + "baz"          = "false"
...
```

I am able to manually update tags in the azure portal.

**Comment by tombuildsstuff** [comment]
hi @shawnferry 

Thanks for opening this issue.

Whilst the Azure Portal supports updating the tags for an App Service Environment, unfortunately at this time the App Service Environment API doesn't support this via the Update API, which is why this field is ForceNew.

From our side when the API supports this we can look to update the Azure Provider to support this - but at this time unfortunately this is blocked on the App Service Environment API supporting this during the update call.

Thanks!

**Comment by shawnferry** [comment]
@tombuildsstuff  gotcha, that makes sense.  thanks. I think I have another issue in the same vein that I'm working through with placing the app env in a resource group.

**Comment by favoretti** [comment]
Since this issue has been reported a long time ago and relates to the version of provider we no longer support - I'm going to close it. Please open a new updated bug report on current versions of terraform and provider if this is still relevant. Thank you.
